src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -36,105 +37,144 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
                 }
                 else
                 {
-                    var shouldRefresh = SilentRequestHelper.NeedsRefresh(cachedAccessTokenItem);
-
-                    // may fire a request to get a new token in the background
-                    if (shouldRefresh)
-                    {
-                        AuthenticationRequestParameters.RequestContext.ApiEvent.CacheInfo = CacheRefreshReason.ProactivelyRefreshed;
-
-                        SilentRequestHelper.ProcessFetchInBackground(
-                        cachedAccessTokenItem,
-                        () => FetchNewAccessTokenAsync(cancellationToken), logger);
-                    }
+                    logger.Verbose(() => "[GetAccessTokenAsync] Getting Access token from cache ...");
+                    authResult = CreateAuthenticationResultFromCache(cachedAccessTokenItem);
                 }
-
+                
                 return authResult;
             }
-            catch (MsalServiceException e)
+            finally
             {
-                return await HandleTokenRefreshErrorAsync(e, cachedAccessTokenItem).ConfigureAwait(false);
+                s_semaphoreSlim.Release();
+                logger.Verbose(() => "[GetAccessTokenAsync] Released token acquire for client credential request semaphore.");
             }
         }
 
-        private async Task<AuthenticationResult> FetchNewAccessTokenAsync(CancellationToken cancellationToken)
+        private async Task<AuthenticationResult> SendTokenRequestToProviderAsync(ILoggerAdapter logger, 
+            CancellationToken cancellationToken)
         {
-            await ResolveAuthorityAsync().ConfigureAwait(false);
-            MsalTokenResponse msalTokenResponse;
-            if (ServiceBundle.Config.AppTokenProvider != null)
-            {
-                msalTokenResponse = await SendTokenRequestToProviderAsync(cancellationToken).ConfigureAwait(false);
-            }
-            else
-            {
-                msalTokenResponse = await SendTokenRequestAsync(GetBodyParameters(), cancellationToken).ConfigureAwait(false);
-            }
+            logger.Info("[GetAccessTokenAsync] Sending Token response to client credential request endpoint ...");
+            MsalTokenResponse msalTokenResponse = await SendRequestToProviderAsync(cancellationToken).ConfigureAwait(false);
 
-            return await CacheTokenResponseAndCreateAuthenticationResultAsync(msalTokenResponse).ConfigureAwait(false);
+            AuthenticationResult authResult = await CacheTokenResponseAndCreateAuthenticationResultAsync(msalTokenResponse)
+                .ConfigureAwait(false);
+
+            return authResult;
         }
 
-        private async Task<MsalTokenResponse> SendTokenRequestToProviderAsync(CancellationToken cancellationToken)
+        private async Task<MsalTokenResponse> SendRequestToProviderAsync(CancellationToken cancellationToken)
```

Comment:
nit: I would merge this method into `SendTokenRequestToProviderAsync`. The naming on the methods is too similar.